src/tests/GC/Stress/Framework/ReliabilityFramework.cs

```diff
@@ -314,6 +322,16 @@ public static int Main(string[] args)
         return (retVal);
     }
 
+    public void RecordTestRanCount()
+    {
+        StringBuilder sb = new();
+        foreach(var item in _testRanCounter)
```

Comment:
When called via the `CancelKeyPress` event, the test will still be running while this code executes.  Use _testRanCounterLock here.